litmus-portal/graphql-server/pkg/myhub/handler/handler.go

```diff
@@ -191,3 +198,140 @@ func IsFileExisting(path string) (bool, error) {
+	}
+	defer destDir.Close()
+
+	//download the zip file from the provided url
+	download, err := http.Get(hubDetails.RepoURL)
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+
+	defer download.Body.Close()
+
+	if download.StatusCode != http.StatusOK {
+		err = fmt.Errorf("err: ", download.Status)
+		return err
+	}
+
+	//validate the content length (in bytes)
+	maxSize, err := strconv.Atoi(os.Getenv("REMOTE_HUB_MAX_SIZE"))
+	if err != nil {
+		return err
+	}
+	contentLength := download.Header.Get("content-length")
+	length, err := strconv.Atoi(contentLength)
+	if length > maxSize {
+		_ = os.Remove(hubpath)
+		err = fmt.Errorf("err: File size exceeded the threshold %d", length)
+		return err
+	}
+
+	//validate the content-type
+	contentType := download.Header.Get("content-type")
+	if contentType != "application/zip" {
+		_ = os.Remove(hubpath)
+		err = fmt.Errorf("err: Invalid file type %s", contentType)
+		return err
+	}
+
+	//copy the downloaded content to the created zip file
+	_, err = io.Copy(destDir, download.Body)
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+
+	//unzip the ChaosHub to the default hub directory
+	err = UnzipRemoteHub(hubpath, hubDetails)
+	if err != nil {
+		return err
+	}
+
+	//remove the redundant zip file
+	err = os.Remove(hubpath)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+//UnzipRemoteHub is used to unzip the zip file
+func UnzipRemoteHub(zipPath string, hubDetails model.CreateRemoteMyHub) error {
+	extractPath := defaultPath + hubDetails.ProjectID
+	zipReader, err := zip.OpenReader(zipPath)
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+	defer zipReader.Close()
+	for _, file := range zipReader.File {
+		CopyZipItems(file, extractPath, file.Name)
+	}
+	return nil
+}
+
+//CopyZipItems is used to copy the content from the extracted zip file to
+//the ChaosHub directory
+func CopyZipItems(file *zip.File, extractPath string, chartsPath string) error {
+	path := filepath.Join(extractPath, chartsPath)
+	if !strings.HasPrefix(path, filepath.Clean(extractPath)+string(os.PathSeparator)) {
+		return fmt.Errorf("illegal file path: %s", path)
+	}
+	err := os.MkdirAll(filepath.Dir(path), os.ModeDir|os.ModePerm)
+	if err != nil {
+		fmt.Println(err)
+	}
+	fileReader, err := file.Open()
+	if err != nil {
+		fmt.Println(err)
+	}
+	if !file.FileInfo().IsDir() {
+		fileCopy, err := os.Create(path)
+		if err != nil {
+			fmt.Println(err)
+		}
+		_, err = io.Copy(fileCopy, fileReader)
+		if err != nil {
+			fmt.Println(err)
+		}
+		fileCopy.Close()
+	}
+	fileReader.Close()
+
+	return nil
+}
+
+//SyncRemoteRepo is used to sync the remote ChaosHub
+func SyncRemoteRepo(hubData model.CloningInput) {
+	hubPath := defaultPath + hubData.ProjectID + "/" + hubData.HubName + "/experiments"
+	updateHub := model.CreateRemoteMyHub{
+		HubName:   hubData.HubName,
+		RepoURL:   hubData.RepoURL,
+		ProjectID: hubData.ProjectID,
+	}
+	_, err := os.ReadDir(hubPath)
+	if err != nil {
+		if os.IsNotExist(err) {
```

Comment:
Do we need these checks? Since this is syncing remote repo, so the folder will already be there.
With this check in place, don't think remote hub will be synced.
It will always be skipped, since the remote hub is always existing.